src/aliases/api/handlers/handlers.go

```diff
@@ -0,0 +1,299 @@
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"sync"
+
+	"github.com/consensys/quorum-key-manager/pkg/errors"
+	jsonutils "github.com/consensys/quorum-key-manager/pkg/json"
+	"github.com/consensys/quorum-key-manager/src/aliases/api/types"
+	aliasent "github.com/consensys/quorum-key-manager/src/aliases/entities"
+	infrahttp "github.com/consensys/quorum-key-manager/src/infra/http"
+	"github.com/gorilla/mux"
+)
+
+type AliasHandler struct {
+	alias aliasent.AliasBackend
+}
+
+func NewAliasHandler(backend aliasent.AliasBackend) *AliasHandler {
+	h := AliasHandler{
+		alias: backend,
+	}
+
+	return &h
+}
+
+func (h *AliasHandler) Register(r *mux.Router) {
+	regRoute := r.PathPrefix("/registries/{registry_name}").Subrouter()
+	regRoute.HandleFunc("", h.deleteRegistry).Methods(http.MethodDelete)
+
+	alRoute := regRoute.PathPrefix("/aliases").Subrouter()
+	alRoute.HandleFunc("", h.listAliases).Methods(http.MethodGet)
+	alRoute.HandleFunc("/{alias_key}", h.createAlias).Methods(http.MethodPost)
+	alRoute.HandleFunc("/{alias_key}", h.getAlias).Methods(http.MethodGet)
+	alRoute.HandleFunc("/{alias_key}", h.updateAlias).Methods(http.MethodPut)
+	alRoute.HandleFunc("/{alias_key}", h.deleteAlias).Methods(http.MethodDelete)
+}
+
+// @Summary Delete a registry
+// @Description Delete a registry and all its keys
+// @Tags registry
+// @Param registry_name path string true "registry identifier"
+// @Param registry_key path string true "registry identifier"
+// @Param request body types.DeleteRegistryRequest true "Delete Registry Request"
+// @Success 204 {object} types.DeleteRegistryResponse "Registry data"
+// @Failure 400 {object} ErrorResponse "Invalid request format"
+// @Failure 500 {object} ErrorResponse "Internal server error"
+// @Router /registryes/{registry_name}/registryes/{registry_key} [delete]
+func (h *AliasHandler) deleteRegistry(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	// should always exists in this subrouter
+	regName := vars["registry_name"]
+
+	err := validatePathVars(regName)
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, errors.InvalidFormatError(err.Error()))
+		return
+	}
+
+	err = h.alias.DeleteRegistry(r.Context(), aliasent.RegistryName(regName))
+	if err != nil {
+		infrahttp.WriteHTTPErrorResponse(w, err)
+		return
+	}
+	w.Header().Set("X-Content-Type-Options", "nosniff")
```

Comment:
What is this header for?

Reply:
I copied it from https://github.com/ConsenSys/quorum-key-manager/commit/7a6e202#diff-5ef92eb3fa1b476f02e830de80aea28c596776a886f710278317a67579f66765

TIL about deprecating the `X-` prefixing, thanks!
But in that case, I thought it was mostly to make IE not try to detect charset from the initial bytes, because it can trip for json and thinks the charset is the default ISO-8859-1, although, for application/json, it is specifically utf-8 by default.

Reply:
@nmvalera said we should also get rid of it.